### PR TITLE
Python changes for adding `__getitem__` for `struct`

### PR DESCRIPTION
**Pull request by shaneding**
Implements necessary `__getitem__` changes for python side. Partly addresses #8558, merge after #8578

### CODE REVIEW

**Review by brandon-b-miller — APPROVED** [review]
I think this looks good. Great job! :)

**Comment by shaneding** [comment]
rerun tests

File: python/cudf/cudf/core/column/struct.py [hunk 1]
```diff
@@ -80,6 +80,15 @@ def to_arrow(self):
             pa_type, len(self), buffers, children=children
         )
 
+    def __getitem__(self, args):
+        result = super().__getitem__(args)
+        if isinstance(result, dict):
+            return dict(
+                (field, value)
+                for field, value in zip(self.dtype.fields, result.values())
+            )
```
Comment (shwina) [inline python/cudf/cudf/core/column/struct.py:89]: dict comprehension is just a little cleaner here:

```suggestion
            return {
                field: value for field, value in zip(self.dtype.fields, result.values())
            }
```

Comment (shwina) [reply python/cudf/cudf/core/column/struct.py:89]: May fail style check so don't accept the suggestion directly :)

File: python/cudf/cudf/core/indexing.py [hunk 2]
```diff
@@ -93,7 +93,8 @@ def __getitem__(self, arg):
         data = self._sr._column[arg]
 
         if (
-            isinstance(data, list)
+            isinstance(data, dict)
+            or isinstance(data, list)
```
Comment (shwina) [inline python/cudf/cudf/core/indexing.py:97]:
```suggestion
            isinstance(data, (dict, list))
```

**Review by shwina — APPROVED** [review]

**Comment by shwina** [comment]
@gpucibot merge

**Pull request merged**